### PR TITLE
Elanjm2/vir 1548 use vitest in ci

### PR DESCRIPTION
**Pull request by elanjm12**

### CODE REVIEW

**Comment by elanjm12** [comment]
Ready for review.

Test files switched to have .jsx ending.
Tests converted where needed to be compatible with vitest.
jest.fn() => vi.fn()
github actions updated

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (igboyes) [inline .vscode/extensions.json]: Remove your `.vscode` directory from the PR. Add the folder to `.gitignore`.

**Review by igboyes — CHANGES_REQUESTED** [review]
What does this mean: `ShallowWrapper {}`? Are you sure our snapshot are still intact?

File: package.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
     "@reduxjs/toolkit": "^1.8.0",
     "@sentry/browser": "^6.18.2",
     "@vitejs/plugin-react": "^2.1.0",
+    "@vue/test-utils": "^2.0.2",
```
Comment (igboyes) [inline package.json]: Should be dev dependencies.

**Review by igboyes — CHANGES_REQUESTED** [review]

**Pull request merged**